src/main/java/network/brightspots/rcv/GuiConfigController.java

```diff
@@ -1135,94 +1205,14 @@ private void setUpHintsForTab(Tab tab, String hintsFilename) {
-            Logger.log(Level.WARNING,
-                "winnerElectionMode \"%s\" is unrecognized! Please supply a valid "
-                    + "winnerElectionMode.", oldWinnerElectionMode);
-            config.rawConfig.rules.winnerElectionMode = null;
-          }
-        }
-      }
-
-      if (config.getTiebreakMode() == TieBreakMode.MODE_UNKNOWN) {
-        Map<String, String> tiebreakModeMigrationMap = Map.of(
-            "random", TieBreakMode.RANDOM.toString(),
-            "interactive", TieBreakMode.INTERACTIVE.toString(),
-            "previousRoundCountsThenRandom",
-            TieBreakMode.PREVIOUS_ROUND_COUNTS_THEN_RANDOM.toString(),
-            "previousRoundCountsThenInteractive",
-            TieBreakMode.PREVIOUS_ROUND_COUNTS_THEN_INTERACTIVE.toString(),
-            "usePermutationInConfig", TieBreakMode.USE_PERMUTATION_IN_CONFIG.toString(),
-            "generatePermutation", TieBreakMode.GENERATE_PERMUTATION.toString()
-        );
-        String oldTiebreakMode = config.rawConfig.rules.tiebreakMode;
-        if (tiebreakModeMigrationMap.containsKey(oldTiebreakMode)) {
-          config.rawConfig.rules.tiebreakMode = tiebreakModeMigrationMap.get(oldTiebreakMode);
-        } else {
-          Logger.log(Level.WARNING,
-              "tiebreakMode \"%s\" is unrecognized! Please supply a valid tiebreakMode.",
-              oldTiebreakMode);
-          config.rawConfig.rules.tiebreakMode = null;
-        }
-      }
-
-      if (config.getOvervoteRule() == OvervoteRule.RULE_UNKNOWN) {
-        String oldOvervoteRule = config.rawConfig.rules.overvoteRule;
-        switch (oldOvervoteRule) {
-          case "alwaysSkipToNextRank" -> config.rawConfig.rules.overvoteRule = OvervoteRule.ALWAYS_SKIP_TO_NEXT_RANK
-              .toString();
-          case "exhaustImmediately" -> config.rawConfig.rules.overvoteRule = OvervoteRule.EXHAUST_IMMEDIATELY
-              .toString();
-          case "exhaustIfMultipleContinuing" -> config.rawConfig.rules.overvoteRule = OvervoteRule.EXHAUST_IF_MULTIPLE_CONTINUING
-              .toString();
-          default -> {
-            Logger.log(Level.WARNING,
-                "overvoteRule \"%s\" is unrecognized! Please supply a valid overvoteRule.",
-                oldOvervoteRule);
-            config.rawConfig.rules.overvoteRule = null;
-          }
-        }
-      }
-
-      Logger.log(
-          Level.INFO,
-          "Migrated tabulator config version from %s to %s.",
-          config.rawConfig.tabulatorVersion != null ? config.rawConfig.tabulatorVersion : "unknown",
-          Main.APP_VERSION);
-      config.rawConfig.tabulatorVersion = Main.APP_VERSION;
-    }
-  }
-
   private void loadConfig(ContestConfig config) {
     clearConfig();
     RawContestConfig rawConfig = config.getRawConfig();
-    migrateConfigVersion(config);
+    try {
+      ContestConfigMigration.migrateConfigVersion(config);
+    } catch (ConfigVersionIsNewerThanAppVersionException e) {
+      return;
```

Comment:
Should this log and re-throw?  Can't do early returns :/

Reply:
fsd;jkfaj;dfskjfdas;kldfsajlk;adfjdfklas